### PR TITLE
"fastmos"

### PR DESCRIPTION
**Pull request by Kmc2000**
## Why It's Good For The Game

What's more important, atmos sim or being able to move?

## Changelog
:cl:
tweak: Atmospherics processing slows down when the ship is in heavy combat.
/:cl:

### CODE REVIEW

**Comment by DeltaFire15** [comment]
I think one of the main issues with atmos breaches is upper deck breaches causing a chain of 'thing gets sucked out through to the top deck -> thing falls down since it's openspace -> repeat forever' (aka, spinny tile of doom)
Might be worth to try disabling cross-z pulling and see if that helps.

**Comment by Kmc2000** [comment]
> I think one of the main issues with atmos breaches is upper deck breaches causing a chain of 'thing gets sucked out through to the top deck -> thing falls down since it's openspace -> repeat forever' (aka, spinny tile of doom)
> Might be worth to try disabling cross-z atmos pulling and see if that helps.

Oh yeah this is just the heavy handed bruteforce way of doing that

**Comment by SerynEngi** [comment]
> I think one of the main issues with atmos breaches is upper deck breaches causing a chain of 'thing gets sucked out through to the top deck -> thing falls down since it's openspace -> repeat forever' (aka, spinny tile of doom)
> Might be worth to try disabling cross-z atmos pulling and see if that helps.

This would also have the perk of making the job of engineers much simpler on maps with huge topdecks like the Galactica or Tycoon.

**Comment by DeltaFire15** [comment]
Yeah, 'not dragging objects up' is what I meant, not 'ceiling holes are no more'

**Comment by TerryTehCoder** [comment]
> Yeah, 'not dragging objects up' is what I meant, not 'ceiling holes are no more'

I did always find this weird, that when objects are sucked out in decompression, they just stay there at the top(and then fall down, since like openspace. But mechanically, there's nowhere else for them to go. 
But, shouldn't decompression launch stuff off into space if it's strong enough? Not just on top of the hull. 

It's just sort of the way Z-levels are done, and excessive work for something tiny like this, but I think being forcefully ejected by decompression up a deck and into space has intense comedic value. But not something that ever works in a proper gameplay loop.

**Comment by DeltaFire15** [comment]
Also this has been mentioned in discord but to leave it here for easier access, but
Just slowing down SSair like this PR does would cause major issues with a bunch of things like both the main engines (SD and AGCNR), due to them ticking on SSmachinery and worsifying the phenomenon of 'hey the engine stopped working because the gas doesn't move in its pipes anymore

**Comment by TheFakeElon** [comment]
prob(80)

File: code/controllers/subsystem/air.dm [hunk 1]
```diff
@@ -84,6 +84,18 @@ SUBSYSTEM_DEF(air)
 /datum/controller/subsystem/air/fire(resumed = 0)
 	var/timer = TICK_USAGE_REAL
 
+	var/TiDi = round((Master.tickdrift/(world.time/world.tick_lag))*100,0.1)
+	switch(TiDi)
+		if(0 to 15)
+			wait = 5 //This isn't an initial() call because that could get nasty with how much this thing runs.
```
Comment (TheFakeElon) [inline code/controllers/subsystem/air.dm:90]: Might be worth making this a define. Could even have the other wait times as multiplications of the define if you're a gamer 😎

**Comment by covertcorvid** [comment]
Closing due to engine concerns.

**Pull request closed without merging**